nmigen/back/rtlil.py

```diff
@@ -843,13 +843,17 @@ def convert_fragment(builder, fragment, name, top):
     # to create a cell for us in the parent module.
     port_map = OrderedDict()
     for signal in fragment.ports:
-        port_map[compiler_state.resolve_curr(signal)] = signal
+        signal.name = compiler_state.resolve_curr(signal)
+        port_map[signal.name] = signal
```

Comment:
That's backwards. The constraint file should be adapted to the signal names, not the other way around.